otter/util/zkpartitioner.py

```diff
@@ -0,0 +1,133 @@
+"""
+ZooKeeper set-partitioning stuff.
+"""
+
+from twisted.application.internet import TimerService
+from twisted.application.service import MultiService
+from twisted.internet.defer import succeed
+
+
+class Partitioner(MultiService, object):
+    """
+    A Twisted service which uses a Kazoo :obj:`SetPartitioner` to allocate
+    logical ``buckets`` between nodes.
+
+    Multiple servers can instantiate these with the same ``partitioner_path``
+    and ``buckets``, and each server will get a disjoint subset of
+    ``range(buckets)`` allocated to them.
+
+    In order to be notified of which buckets are allocated to the current node,
+    a ``got_buckets`` function must be passed in which will be called when the
+    local buckets have been determined.
+    """
+    def __init__(self, kz_client, interval, partitioner_path, buckets,
+                 time_boundary, log, got_buckets,
+                 clock=None):
+        """
+        :param log: a bound log
+        :param kz_client: txKazoo client
+        :param partitioner_path: ZooKeeper path, used for partitioning
+        :param list buckets: buckets to distribute between nodes. Ideally
```

Comment:
This should be `param set buckets`. Also, I didn't understand why this should be `sequence of str`